servicemanager/servicemanager.go

```diff
@@ -134,9 +194,41 @@ func (sm *ServiceManager) InstallService(
 	)
 }
 
+// RemoveService remove service from the system.
+func (sm *ServiceManager) RemoveService(serviceID string) (err error) {
+	return <-sm.actionHandler.Execute(serviceID,
+		func(id string) error {
+			return sm.doRemoveService(serviceID)
+		},
+	)
+}
+
+// TryAllocateMemory try allocate memory by remove outdate services.
+func (sm *ServiceManager) TryAllocateMemory(
+	availableSize, requiredSize int64,
+) (allocatedLayersSize int64, err error) {
+	err = <-sm.actionHandler.Execute(strconv.FormatInt(availableSize+requiredSize, 10),
+		func(id string) error {
+			if allocatedLayersSize, err = sm.removeOutdateService(availableSize, requiredSize); err != nil {
```

Comment:
I guess we should use here same function as in tryToAllocateSpace, namely, we should delete pending remove services only, and only the required amount of size.

Reply:
Done